src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -2,21 +2,49 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
+import mate.academy.hibernate.relations.lib.Dao;
+import mate.academy.hibernate.relations.lib.Factory;
 import mate.academy.hibernate.relations.model.Actor;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
+@Dao
 public class ActorDaoImpl extends AbstractDao implements ActorDao {
+    @Factory
     public ActorDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add actor " + actor);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't add actor " + actor, e);
```

It is better to add the old exception so that it is not lost. And so the original reason will not be clear.

Reply:
Ok, thanks